src/web/lib/hakuneko/frontend@modern/components/menuComponent/script.js

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-vars */
+class HakunekoMenu extends Polymer.Element {
+    static get is() {
+        return "hakuneko-menu";
+    }
+
+    static get properties() {
+        return {};
+    }
+
+    ready() {
+        super.ready();
+        this.year = new Date().getFullYear();
+        this.version = Engine.Version;
+        this.popupVisibility = false;
+        this.popupVisibilityAppearance = false;
+        this.settingCategories = Engine.Settings.getCategorizedSettings();
+        this.corners = [
+            {
+                label: this.i18n('settings.corners.straight'),
+                class: "square",
+                value: "corner__square",
+            },
+            {
+                label: this.i18n('settings.corners.rounded'),
+                class: "rounded",
+                value: "corner__rounded",
+            },
+        ];
+
+        this.themes = [
+            { label: this.i18n('settings.colors.red'), value: "red", color: "#F44336" },
+            { label: this.i18n('settings.colors.pink'), value: "pink", color: "#E91E63" },
+            { label: this.i18n('settings.colors.purple'), value: "purple", color: "#9C27B0" },
+            { label: this.i18n('settings.colors.dpurple'), value: "dpurple", color: "#673AB7" },
+            { label: this.i18n('settings.colors.indigo'), value: "indigo", color: "#3F51B5" },
+            { label: this.i18n('settings.colors.blue'), value: "blue", color: "#2196F3" },
+            { label: this.i18n('settings.colors.cian'), value: "cian", color: "#00BCD4" },
+            { label: this.i18n('settings.colors.teal'), value: "teal", color: "#009688" },
+            { label: this.i18n('settings.colors.green'), value: "green", color: "#4CAF50" },
+            { label: this.i18n('settings.colors.lgreen'), value: "lgreen", color: "#8BC34A" },
+            { label: this.i18n('settings.colors.lime'), value: "lime", color: "#CDDC39" },
+            { label: this.i18n('settings.colors.yellow'), value: "yellow", color: "#FFEB3B" },
+            { label: this.i18n('settings.colors.amber'), value: "amber", color: "#FFC107" },
+            { label: this.i18n('settings.colors.orange'), value: "orange", color: "#FF9800" },
+            { label: this.i18n('settings.colors.dorange'), value: "dorange", color: "#FF5722" },
+        ];
+
+        this.languages = [
+            { label: this.i18n('settings.languages.english'), value: 'en'},
+            { label: this.i18n('settings.languages.spanish'), value: 'es'},
+        ]
+    }
+
+    getLanguageSelected(value) {
+        return localStorage.getItem("lang") === value ? "selected" : "";
+    }
+
+    getCornerSelected(value) {
+        return localStorage.getItem("corner") === value ? "selected" : "";
```

Comment:
Key name is to common for global scope, use something like `frontend.modern.corner` instead of `corner`
Also be aware that local storage is not persistent, it can be wiped randomly by the chrome engine